File: client/ios/UseCase/System/Push/RegisterPushTokenUseCase.swift
import Foundation

public struct RegisterPushTokenUseCase {
    private let pushTokenRegistrar: PushTokenRegistrar

    public init(pushTokenRegistrar: PushTokenRegistrar) {
        self.pushTokenRegistrar = pushTokenRegistrar
    }

    public func callAsFunction(
        token: String,
        platform: String = "ios",
        appKey: String,
        deviceId: String,
        enabled: Bool = true
    ) async throws -> Bool {
        try await pushTokenRegistrar.register(
            payload: PushRegistrationPayload(token: token, platform: platform, appKey: appKey, deviceId: deviceId, enabled: enabled)
        )
    }
}

File: client/android/UseCase/System/Push/RegisterPushTokenUseCase.kt
package app.mobiling.client.usecase.system.push

import app.mobiling.client.contract.system.push.PushRegistrationPayload
import app.mobiling.client.data.system.push.PushTokenRegistrar

class RegisterPushTokenUseCase(private val pushTokenRegistrar: PushTokenRegistrar) {
    operator fun invoke(
        token: String,
        platform: String = "android",
        appKey: String,
        deviceId: String,
        enabled: Boolean = true,
    ): Boolean = pushTokenRegistrar.register(
        PushRegistrationPayload(token = token, platform = platform, appKey = appKey, deviceId = deviceId, enabled = enabled),
    )
}
